control-plane/api/v1alpha1/api_gateway_types.go

```diff
@@ -0,0 +1,260 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package v1alpha1
+
+import (
+	"fmt"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+const (
+	GatewayClassConfigKind = "GatewayClassConfig"
+	managedLabel           = "api-gateway.consul.hashicorp.com/managed"
+	nameLabel              = "api-gateway.consul.hashicorp.com/name"
+	namespaceLabel         = "api-gateway.consul.hashicorp.com/namespace"
+	createdAtLabel         = "api-gateway.consul.hashicorp.com/created"
+)
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster
+
+// GatewayClassConfig defines the values that may be set on a GatewayClass for Consul API Gateway.
+type GatewayClassConfig struct {
+	// Standard Kubernetes resource metadata.
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of GatewayClassConfig.
+	Spec GatewayClassConfigSpec `json:"spec,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+
+// GatewayClassConfigSpec specifies the desired state of the Config CRD.
+type GatewayClassConfigSpec struct {
+	// +kubebuilder:validation:Enum=ClusterIP;NodePort;LoadBalancer
+	ServiceType *corev1.ServiceType `json:"serviceType,omitempty"`
+
+	// NodeSelector is a selector which must be true for the pod to fit on a node.
+	// Selector which must match a node's labels for the pod to be scheduled on that node.
+	// More info: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// Tolerations allow the scheduler to schedule nodes with matching taints.
+	// More Info: https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// Annotation Information to copy to services or deployments
+	CopyAnnotations CopyAnnotationsSpec `json:"copyAnnotations,omitempty"`
+}
+
+// PortSpec defines the ports that the API Gateway will listen on.
```

Comment:
This seems to be unused, as it was formerly part of the Consul connection config which was stripped out (due to the fact that we should probably inline it to the deployment the controller creates to pass values from helm directly, similar to things like the connect injection hooks)